packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -165,9 +181,24 @@ class CodeSizeController {
 
       diffMap = compareProgramInfo(oldApkProgramInfo, newApkProgramInfo);
     } else {
-      diffMap = buildComparisonTreemap(oldFile.data, newFile.data);
+      try {
+        diffMap = buildComparisonTreemap(oldFile.data, newFile.data);
+      } catch (error) {
+        onError('Failed to load snapshot: File type not supported.');
+        return;
+      }
+    }
+
+    if ((diffMap['children'] as List).isEmpty) {
+      onError(
+        'Failed to load snapshot: Given snapshots to compare are identical.',
```

Comment:
"Failed to load diff: OLD and NEW files are identical."

Reply:
Done